flexget/plugins/list/couchpotato_list.py

```diff
@@ -0,0 +1,220 @@
+                    for profile in profile_json['list']:
+                        if profile['_id'] == movie['profile_id']:  # Matches movie profile with profile JSON
+                            quality_req = CouchPotatoBase.quality_requirement_builder(profile)
+                entry = Entry(title=movie["title"],
+                              url='',
+                              imdb_id=movie['info'].get('imdb'),
+                              tmdb_id=movie['info'].get('tmdb_id'),
+                              quality_req=quality_req,
+                              couchpotato_id=movie.get('_id'))
+                if entry.isvalid():
+                    log.debug('adding entry %s', entry)
+                    entries.append(entry)
+                else:
+                    log.error('Invalid entry created? %s', entry)
+                    continue
+                # Test mode logging
+                if entry and test_mode:
+                    log.info("Test mode. Entry includes:")
+                    for key, value in entry.items():
+                        log.info('     %s: %s', (key.capitalize(), value))
+
+        return entries
+
+    @staticmethod
+    def add_movie(config, entry, test_mode=None):
+        if not entry.get('imdb_id'):
+            log.error('Cannot add movie to couchpotato without and imdb ID: %s',  entry)
+            return
+        log.verbose('Connection to CouchPotato to add a movie to list.')
+        add_movie_url = CouchPotatoBase.build_url(config.get('base_url'), 'add', config.get('port'),
+                                                  config.get('api_key'))
+        title = entry.get('movie_name')
+        imdb_id = entry.get('imdb_id')
+        add_movie_url += '?title=%s&identifier=%s' % (title, imdb_id)
+        add_movie_json = CouchPotatoBase.get_json(add_movie_url)
+        return add_movie_json['movie']
+
+    @staticmethod
+    def remove_movie(config, movie_id, test_mode=None):
+        log.verbose('Deleting movie from Couchpotato')
+        delete_movie_url = CouchPotatoBase.build_url(config.get('base_url'), 'delete', config.get('port'),
+                                                     config.get('api_key'))
+        delete_movie_url += '&id=%s' % movie_id
+        CouchPotatoBase.get_json(delete_movie_url)
+
+
+class CouchPotatoList(MutableSet):
+    def _find_entry(self, entry):
+        for cp_entry in self.movies:
+            # TODO This could be a little too broad, rethink this
+            for key in entry:
+                if entry[key] == cp_entry[key]:
+                    return True
+
+    def __init__(self, config):
+        self.config = config
+        self.movies = CouchPotatoBase.list_entries(config)
+
+    def __iter__(self):
+        return (entry for entry in self.movies)
+
+    def __len__(self):
+        return len(self.movies)
+
+    def __contains__(self, entry):
+        return self._find_entry(entry) is not None
+
+    def __ior__(self, other):
```

Comment:
The ABC should give us this one for free. Only need to override it if we need to customize how it works.